packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx

```diff
@@ -1,162 +1,232 @@
 import { useUser } from '@app/hooks/useUser';
-import { ActivityHistoryItem, BountyClaimCollection, BountyCollection, StatusHistoryItem } from '@app/models/Bounty';
+import {
+	ActivityHistoryItem,
+	BountyClaimCollection,
+	BountyCollection,
+	StatusHistoryItem,
+} from '@app/models/Bounty';
 import axios from '@app/utils/AxiosUtils';
-import { Alert, AlertIcon, Button, Flex, Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay, Text, Textarea, Tooltip, useColorMode, useDisclosure } from '@chakra-ui/react';
-import { useRouter } from 'next/router';
+import {
+	Alert,
+	AlertIcon,
+	Button,
+	Box,
+	Flex,
+	Modal,
+	ModalBody,
+	ModalCloseButton,
+	ModalContent,
+	ModalFooter,
+	ModalHeader,
+	ModalOverlay,
+	Text,
+	Textarea,
+	Tooltip,
+	useColorMode,
+	useDisclosure,
+} from '@chakra-ui/react';
+// import { useRouter } from 'next/router';
 import { useState } from 'react';
 import { useContext } from 'react';
-import { claimedBy, newActivityHistory, newStatusHistory } from '@app/utils/formUtils';
+import {
+	claimedBy,
+	isClaimableByUser,
+	newActivityHistory,
+	newStatusHistory,
+} from '@app/utils/formUtils';
 import AccessibleLink from '@app/components/parts/AccessibleLink';
 import { CustomerContext } from '@app/context/CustomerContext';
 import { baseUrl } from '@app/constants/discordInfo';
 import { useRequiredRoles } from '@app/components/global/Auth';
 import { mutate } from 'swr';
 import BOUNTY_STATUS from '@app/constants/bountyStatus';
 import ACTIVITY from '@app/constants/activity';
+import miscUtils from '@app/utils/miscUtils';
+import { APIUser } from 'discord-api-types';
 
 const BountyClaim = ({ bounty }: { bounty: BountyCollection }): JSX.Element => {
 	const { isOpen, onOpen, onClose } = useDisclosure();
-	const router = useRouter();
+	// const router = useRouter();
 	const { colorMode } = useColorMode();
 	const { user } = useUser();
 	const [message, setMessage] = useState<string>();
-	const [claiming, setClaiming] = useState(false);
+	const [claiming, setClaiming] = miscUtils.useStateCallback<boolean>(false);
```

Comment:
We can just use the useState here. Why are we using useStateCallback here?

Reply:
I’m not the author of that particular line

Reply:
Only want the modal box to close if there is no error, otherwise leave it open. So callback to onClose is used.